File: dan/Problems/Easy/Array/66. Plus One/solution.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        digits[-1] += 1
        for i in range(len(digits) - 1, -1, -1):
            if digits[i] < 10:
                return digits
            digits[i] -= 10
            if i > 0:
                digits[i - 1] += 1
            else:
                return [1] + digits
            

File: dan/Problems/Easy/Array/66. Plus One/solution.java
class Solution {
    public int[] plusOne(int[] digits) {
        int flag = 0;
        for(int i = digits.length - 1; i >= 0; --i){
            if(digits[i] < 9){
                digits[i]++;
                flag = 1;
                break;
            }else{
                digits[i] = 0;
            }
        }
        int[] list = new int[digits.length + 1];
        list[0] = 1;
        return (flag == 1)?digits:list;
        
    }
}
